ecl/eclagent/eclagent.cpp

```diff
@@ -605,12 +614,20 @@ const char *EclAgent::queryTempfilePath()
                 dir.append(path);
             else
                 dir.append("c:");
-            dir.append("\\HPCCSystems\\hthortemp");
+            dir.append("\\HPCCSystems\\").append(instance.str());
 #else
-            dir.append("/tmp/HPCCSystems/hthortemp");
+            dir.append(getenv("TMPDIR"));
+            if (!dir.length())
+                dir.append("/tmp");
```

Comment:
Change looks ok.  A couple of questions:
- Why not use RUNTIME_DIR as you mentioned in the jira.
- Why the switch to instance.str() instead of hthortemp
Might there be a problem if the temp files get very large?

Reply:
RUNTIME_DIR suffers the same issue, that it is owned by hpcc, and so if TMPDIR is not set then it results in the same permission failure.
In this use case the instance is empty and I thought it a bit more consistent to use the instance if present and use hthortemp if it is empty.